js/walutomat.js

```diff
@@ -0,0 +1,270 @@
+        const result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const symbolObject = { 'symbol': this.marketId (symbol) };
+        const response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'baseVolume');
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'operationType': 'MARKET_FX',
+            'sortOrder': 'DESC',
+        };
+        if (symbol !== undefined) {
+            request['currencies'] = this.marketId (symbol);
+        }
+        if (since !== undefined) {
+            // TODO: Not sure if it's correct (idk if it's timestamp or DateTime string
+            request['continueFrom'] = since;
+        }
+        if (limit !== undefined) {
+            request['volume'] = limit;
+        }
+        const response = await this.privateGetAccountHistory (this.extend (request, params));
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
+            result.push ({
+                'id': this.safeInteger (response[i], 'id'),
+            });
+        }
+        return result;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const body = {
+            'submitId': this.uuid (),
+            'pair': market['id'],
+            'price': price,
+            'buySell': side.toUpperCase (),
+            'volumeCurrency': market['baseId'],
+            'otherCurrency': market['quoteId'],
+            'volume': amount,
+        };
+        return await this.privatePostMarketOrders (this.extend (body, params));
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        const request = {
+            'orderId': id,
+        };
+        return await this.privatePostMarketOrdersCloseOrderId (this.extend (request, params));
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // TODO: Would be nice to get parametrized API response
+        await this.loadMarkets ();
+        const orders = await this.privateGetMarketOrders ();
+        const result = [];
+        for (let i = 0; i < orders.length; i++) {
+            result.push ({
```

Comment:
This loop has the same issue as with fetchMyTrades – we have `parseOrders` in the base class and `parseOrder` should be defined in the derived class here. You practically have the body of `parseOrder` implemented inside `fetchOrders` – can you plz detach it to a standalone `parseOrder` and change this loop for a call to `parseOrders` instead ?